core/src/main/java/edu/wpi/grip/core/Palette.java

```diff
@@ -4,26 +4,28 @@
 import com.google.common.eventbus.Subscribe;
 import edu.wpi.grip.core.events.OperationAddedEvent;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
 import java.util.Collection;
-import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Optional;
 
-import static org.python.jline.internal.Preconditions.checkNotNull;
-
 /**
  * The palette is a library of operations that can be added as steps in the {@link Pipeline}
  */
+@Singleton
 public class Palette {
 
     private final EventBus eventBus;
-    private final Map<String, Operation> operations = new HashMap<>();
 
-    public Palette(EventBus eventBus) {
-        this.eventBus = checkNotNull(eventBus);
-        eventBus.register(this);
+    @Inject
```

Comment:
Used in the [PaletteTest](https://github.com/WPIRoboticsProjects/GRIP/pull/215/files#diff-1299c69c8500e1bd66d49533429edf4cR22)